src/som/interpreter/actors/EventualMessage.java

```diff
@@ -281,7 +308,7 @@ protected PromiseSendMessage(final SSymbol selector, final Object[] arguments,
     /**
      * The promise on which this callback is registered on.
      */
-    protected final SPromise promise;
+    protected SPromise promise;
```

Comment:
Could you at least make it `@CompilationFinal` then? This is likely a performance regression.